### PR TITLE
Feature/gsagive 136 - HMAC digital signature generation

### PR DESCRIPTION
**Pull request by pshumateEDC**
Add javacsript file containing code for the generation of a hmac digital signature that complies with Kong's specifications. Adjust the readme file to describe the process of using it accordingly.

### CODE REVIEW

File: post-deployment.sh [hunk 1]
```diff
@@ -1,6 +1,5 @@
 #!/bin/bash
 echo 'post deployment script'
-
 cf ssh -N -T -L 80:localhost:8081 give-api-gateway &
-curl -X POST http://localhost:80/plugins/ --data "name=hmac-auth" 
+curl -X POST http://localhost:80/plugins/ --data "name=hmac-auth" -d "config.enforce_headers=date, request-line" &
```
Comment (pkothare) [inline post-deployment.sh:4]: While this configures HMAC Auth, we also need an integration test as a confirming step, which is not present in the pull request. Please add the test accordingly.

The test should fetch the current configuration for the deployed instance and ensure that the desired HMAC configuration is present in the JSON.

**Review by pkothare — CHANGES_REQUESTED** [review]
Needs an integration test and slight restructuring of the README contents.

File: README.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 - [Git Bash](https://git-scm.com/downloads)
 - [CF CLI](https://easydynamics.atlassian.net/wiki/spaces/GSATTS/pages/1252032607/Cloud.gov+CF+CLI+Setup)
 - Cloud.gov account (Contact [Will Shah](mailto:wshah@easydynamics.com?subject=GSA%20Cloud.gov%20Account) to get one).
+- [Postman](https://www.postman.com/downloads/)
```
[the remark this answers is not in the document]
Comment (pshumateEDC) [reply README.md]: Removed Postman references from README

File: README.md [hunk 3]
```diff
@@ -40,6 +41,19 @@ cf logs <your-gateway-name>
 ```
 to tail the logs. Include `--recent` if you want to just dump the logs instead.
 
+### HMAC Authentication
+
+The API Gateway is secured with Kong's HMAC Authentication plugin. A digital signature is required in the header of a request that complies with the [Kong HMAC signature specification](https://docs.konghq.com/hub/kong-inc/hmac-auth/#signature-string-construction).
+
+### Making an Authenticated Request (in Postman)
```
[the remark this answers is not in the document]
Comment (pshumateEDC) [reply README.md]: New README exists in `tests/`

File: post-deployment.sh (hunk 1 above)
Comment (pshumateEDC) [reply post-deployment.sh:4]: Python integration tests now exist in `tests/`.

**Comment by pshumateEDC** [comment]
Current state of application in cloud foundry:
```
11:15:07.509: [STG.0] -----> Adding apt repos
11:15:07.509: [STG.0] Added repo {deb https://kong.bintray.com/kong-deb bionic main }
11:15:07.509: [STG.0] -----> Updating apt cache
11:18:31.805: [STG.0]        **ERROR** Error running supply: failed to apt-get update Ign:1 https://kong.bintray.com/kong-deb bionic InRelease
11:18:31.805: [STG.0]        Get:2 https://kong.bintray.com/kong-deb bionic Release [3,524 B]
11:18:31.805: [STG.0]        Get:3 https://kong.bintray.com/kong-deb bionic Release.gpg [821 B]
11:18:31.805: [STG.0]        Get:4 https://kong.bintray.com/kong-deb bionic/main amd64 Packages [4,210 B]
11:18:31.805: [STG.0]        Err:4 https://kong.bintray.com/kong-deb bionic/main amd64 Packages
11:18:31.805: [STG.0]          File has unexpected size (4294 != 4210). Mirror sync in progress? [(IP hidden)]
11:18:31.805: [STG.0]          Hashes of expected file:
11:18:31.805: [STG.0]           - Filesize:4210 [weak]
11:18:31.805: [STG.0]           - SHA256:054ea01d5de35adf9c8aa65559c2d9cb2f40349e70a8ddd0ad625fe1f18e6ba1
11:18:31.805: [STG.0]           - SHA1:c80b403a6d6f74216d98854b1d9698c2051b461f [weak]
11:18:31.805: [STG.0]           - MD5Sum:0e1f7ea048f175a83335c8a19790bee9 [weak]
11:18:31.805: [STG.0]          Release file created at: Thu, 01 Oct 2020 14:34:21 +0000
11:18:31.805: [STG.0]        Get:5 http://archive.ubuntu.com/ubuntu bionic InRelease [242 kB]
11:18:31.805: [STG.0]        Get:6 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]
11:18:31.805: [STG.0]        Get:7 http://archive.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]
11:18:31.805: [STG.0]        Get:8 http://archive.ubuntu.com/ubuntu bionic/main amd64 Packages [1,344 kB]
11:18:31.805: [STG.0]        Get:9 http://archive.ubuntu.com/ubuntu bionic/multiverse amd64 Packages [186 kB]
11:18:31.805: [STG.0]        Get:10 http://archive.ubuntu.com/ubuntu bionic/universe amd64 Packages [11.3 MB]
11:18:31.805: [STG.0]        Get:11 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [2,112 kB]
11:18:31.805: [STG.0]        Get:12 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages [45.5 kB]
11:18:31.805: [STG.0]        Get:13 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [2,150 kB]
11:18:31.805: [STG.0]        Get:14 http://archive.ubuntu.com/ubuntu bionic-security/multiverse amd64 Packages [15.0 kB]
11:18:31.805: [STG.0]        Get:15 http://archive.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [1,348 kB]
11:18:31.805: [STG.0]        Get:16 http://archive.ubuntu.com/ubuntu bionic-security/main amd64 Packages [1,733 kB]
11:18:31.805: [STG.0]        rm: cannot remove '/var/cache/apt/archives/partial/*.deb': Permission denied
11:18:31.805: [STG.0]        Fetched 20.7 MB in 3min 23s (102 kB/s)
11:18:31.805: [STG.0]        Reading package lists...
11:18:31.805: [STG.0]        E: Failed to fetch https://kong.bintray.com/kong-deb/dists/bionic/main/binary-amd64/Packages.gz  File has unexpected size (4294 != 4210). Mirror sync in progress? [IP: 52.88.32.158 443]
11:18:31.805: [STG.0]           Hashes of expected file:
11:18:31.805: [STG.0]            - Filesize:4210 [weak]
11:18:31.805: [STG.0]            - SHA256:054ea01d5de35adf9c8aa65559c2d9cb2f40349e70a8ddd0ad625fe1f18e6ba1
11:18:31.805: [STG.0]            - SHA1:c80b403a6d6f74216d98854b1d9698c2051b461f [weak]
11:18:31.805: [STG.0]            - MD5Sum:0e1f7ea048f175a83335c8a19790bee9 [weak]
11:18:31.805: [STG.0]           Release file created at: Thu, 01 Oct 2020 14:34:21 +0000
11:18:31.805: [STG.0]        E: Some index files failed to download. They have been ignored, or old ones used instead.
11:18:31.805: [STG.0]        
11:18:31.805: [STG.0]        
11:18:31.805: [STG.0]        exit status 100
```

**Pull request closed without merging**